linked_list.py

```diff
@@ -0,0 +1,348 @@
+
+# Defines a node in the singly linked list
+class Node(object):
+    def __init__(self, value=None, next_node=None):
+        self.__data = None  # protected
+        self.next_node = next_node
+        self.__set_data_once(value)
+
+    def get_data(self):
+        if self.__data:
+            return self.__data
+        else:
+            return "Undefined"
+
+    def __set_data_once(self, value):

```

Comment:
This method is not needed. Since data should be kept private - write once, after that you can get but not set.